### PR TITLE
JsonWhoIs - fixed error not returned from the integration

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25306

## Screenshots
image

## Does it break backward compatibility?
   - [ ] No

## Must have
- [x] Tests

### CODE REVIEW

File: Packs/JsonWhoIs/Integrations/JsonWhoIs/JsonWhoIs.py [hunk 1]
```diff
@@ -94,7 +94,8 @@ def whois(url: str) -> tuple:
     raw = http_request(method='GET',
                        url_suffix='/api/v1/whois',
                        params=params)
-
+    if 'error' in raw:
```
Comment (Itay4) [inline Packs/JsonWhoIs/Integrations/JsonWhoIs/JsonWhoIs.py:97]: lets do it in `http_request`

File: Packs/JsonWhoIs/Integrations/JsonWhoIs/JsonWhoIs.py [hunk 2]
```diff
@@ -94,7 +94,8 @@ def whois(url: str) -> tuple:
...
+        return_error(f'Error from JsonWhoIS: {raw["error"]}')
```
Comment (Itay4) [inline Packs/JsonWhoIs/Integrations/JsonWhoIs/JsonWhoIs.py:98]:
```suggestion
        raise Exception(f'Error from JsonWhoIS: {raw.get("error")}')
```

File: Packs/JsonWhoIs/Integrations/JsonWhoIs/JsonWhoIs.py [hunk 3]
```diff
@@ -47,7 +47,10 @@ def http_request(method, url_suffix, params=None, max_retry=3):
             break
         if trial == max_retry:  # type: ignore
             raise Exception(f'Error enrich url with JsonWhoIS API, status code {res.status_code}')  # type: ignore
-    return res.json()  # type: ignore
+    raw = res.json()  # type: ignore
```
Comment (Itay4) [inline Packs/JsonWhoIs/Integrations/JsonWhoIs/JsonWhoIs.py:50]: wrap in try-except

File: Packs/JsonWhoIs/Integrations/JsonWhoIs/JsonWhoIs.py [hunk 4]
```diff
@@ -47,7 +47,13 @@ def http_request(method, url_suffix, params=None, max_retry=3):
             break
         if trial == max_retry:  # type: ignore
             raise Exception(f'Error enrich url with JsonWhoIS API, status code {res.status_code}')  # type: ignore
-    return res.json()  # type: ignore
+    try:
+        raw = res.json()  # type: ignore
+    except ValueError:
+        raise DemistoException('Error from JsonWhoIs: Could not parse JSON from response.')
```
Comment (Itay4) [inline Packs/JsonWhoIs/Integrations/JsonWhoIs/JsonWhoIs.py:53]:
```suggestion
        raise DemistoException(f'Error from JsonWhoIs: Could not parse JSON from response. {res.text}')
```

**Comment by kirbles19** [comment]
@jochman Done.

File: Packs/JsonWhoIs/Integrations/JsonWhoIs/JsonWhoIs.py [hunk 5]
```diff
@@ -47,7 +47,15 @@ def http_request(method, url_suffix, params=None, max_retry=3):
             break
         if trial == max_retry:  # type: ignore
             raise Exception(f'Error enrich url with JsonWhoIS API, status code {res.status_code}')  # type: ignore
-    return res.json()  # type: ignore
+    if res is None:
+        raise DemistoException('Error from JsonWhoIs: Could not get a result from the API.')
+    try:
+        raw = res.json()  # type: ignore
```
Comment (Itay4) [inline Packs/JsonWhoIs/Integrations/JsonWhoIs/JsonWhoIs.py:53]: lets set the error code here

**Pull request merged**